### ISSUE TITLE
Option to have Segment for more than 1 site but not all sites. 

### ISSUE THREAD

**Issue by Birkaransingh94** [piwik/piwik#20972]
At the moment, we can only share Segments to all sites or to one. Is there an option to share the segments to more than one site but not all? This would help to reduce the amount of segments.

Raising a feature request for the same.

**Comment by atom-box** [comment]
A large customer asked for this today also.

> "No possibility to mix several sites or segments on one dashboard"

**Comment by Stan-vw** [comment]
@atom-box mixing several sites is possible using a roll-up. Not sure what that short quote means for "mixing several segments on one dashboard", but it sounds different than this ticket. Keen to read a bit more detail about the customer request.

**Comment by heurteph-ei** [comment]
Hi @Stan-vw, you're right, then I think a new feature request should be created...
The need is the possibility to create a dashboard that could display graphs from several sites. A little like https://demo.matomo.cloud/index.php?module=MultiSites&action=index&idSite=1&period=day&date=yesterday, but with dashboard capabilities (all kind of widgets). I don't ask for a roll-up, we would like for example display the *Visit Over Time* widget for site 1, site 2 and site 3 (then with 3 widgets at the same time on screen, one for each site).

**Comment by Stan-vw** [comment]
Right, just to check if I understand: you would like to be able to select MULTIPLE websites, and show their SEPARATE data in the same chart. A bit like the following, but for website metrics:
![image](https://github.com/matomo-org/matomo/assets/52564377/a98b6c03-ddd3-4238-9930-6452a6933dee)

Is that correct?

**Comment by heurteph-ei** [comment]
@Stan-vw The initial need was not necessary displaying them in the same chart, it was displaying in 2 separate charts in the same dashboard (visits over time of site 1 in chart 1, and visits over time of site 2 in chart 2). But your idea is also a good idea!

**Comment by sgiehl** [comment]
We already have a mechanism to compare things. Maybe we could simply allow comparing websites, like it is possible for segments and periods. The only problem I see with that is that some reports might not be able to handle that, like custom dimensions or goals. And it could also be problematic when comparing websites and segments, as certain segments might not be available for all websites...

**Comment by mattab** [comment]
Yes, it would be helpful to be able to share a segment across other websites, selected in a list, rather than choose between "One" website and "All websites". Sometimes people need to create the same segment across many websites which is time consuming both for creating the segments, and keeping them in sync in the future.

In general this would reduce CPU consumption and help Matomo Cloud-hosted users to not reach the segment quotas as quickly (or save time managing many duplicated segments). (copied from my message in https://github.com/matomo-org/matomo/issues/16256)

**Comment by Stan-vw** [comment]
Thanks Matt, for indeed bringing this back to the original ask of the ticket. The other topics are better suited in separate tickets 👍 

Also thanks for explaining that this could create a reduction of CPU consumption, as well as experience that a lot of users currently spend time duplicating segments to multiple (but not all websites).

We probably need to design how one might select multiple (but not all websites) in this dropdown element:
![Screen Shot 2023-12-15 at 11 37 39 AM](https://github.com/matomo-org/matomo/assets/52564377/7eda3b96-7a6b-411e-a4e0-9329172d9649)
@Javi-Ormaechea

**Comment by Stan-vw** [comment]
It might also help with the maximum amount of segments that people might be bumping into if they need to create duplicate segments:
![Screen Shot 2023-12-15 at 11 52 02 AM](https://github.com/matomo-org/matomo/assets/52564377/e63d51c1-29d0-4851-b963-77a72291594e)
